middleware/kubernetes/controller.go

```diff
@@ -35,6 +35,14 @@ func (s *storeToNamespaceLister) List() (ns api.NamespaceList, err error) {
 	return ns, nil
 }
 
+type dnsControllerItf interface {
```

Comment:
Naming: Drop the Itf and just leave it as `dnsController` and rename the type below to `dnsControl`